underscore.js

```diff
@@ -449,8 +449,18 @@
 
   // Return the number of elements in an object.
   _.size = function(obj) {
+    var typeObj = toString.call(obj);
     if (obj == null) return 0;
-    return isArrayLike(obj) ? obj.length : _.keys(obj).length;
+    if (isArrayLike(obj)) {
+      return obj.length;
+    } else {
+      switch (typeObj) {
+        case '[object Set]':
```

Comment:
This use `toString.call(obj)` to get the type.
It is more optimal than using "_.isMap" or "_.isSet".